instrumentation/commons-dbcp2/src/main/java/com/splunk/opentelemetry/commonsdbcp2/DataSourceMetrics.java

```diff
@@ -41,15 +45,29 @@
   /** The maximum number of open connections allowed. */
   private static final String CONNECTIONS_MAX = "db.pool.connections.max";
 
+  // a weak map does not make sense here because each Meter holds a reference to the dataSource
+  private static final Map<BasicDataSourceMXBean, List<Meter>> dataSourceMetrics =
```

Comment:
This could behave in unexpected ways when there is a `BasicDataSourceMXBean` implementation that defines `equals` and `hashCode`. Idk whether this is something we should care about as probably `BasicDataSource` is the only implementation and it does not have `equals` and `hashCode`. `Collections.synchronizedMap(new IdentityHashMap<>())` would be the safe choice. I'll leave it to you to decide whether to change this or not.

Reply:
> Idk whether this is something we should care about as probably BasicDataSource is the only implementation and it does not have equals and hashCode. 

Exactly this ^^

I'll add the comment explaining why it's safe to use a plain `ConcurrentHashMap` though